app-backend/datamodel/src/main/scala/com/azavea/rf/datamodel/ColorCorrect.scala

```diff
@@ -153,28 +153,78 @@ object ColorCorrect {
     val maybeEqualize =
       if (params.equalize) Some(HistogramEqualization(_: MultibandTile, rgbHist)) else None
 
+    case class LayerClipping(redMin: Int, redMax: Int, greenMin: Int, greenMax:Int, blueMin: Int, blueMax: Int)
+    sealed trait ClipValue
+    case class ClipBounds(min: Int, max: Int) extends ClipValue
+    case class MaybeClipBounds(maybeMin: Option[Int], maybeMax: Option[Int]) extends ClipValue
+    case class ClippingParams(band: Int, bounds: ClipValue)
+   
+    val layerRgbClipping = (for {
+      (r :: g :: b :: xs) <- Some(rgbHist.toList)
+      isCorrected   <- Some((r :: g :: b :: Nil).foldLeft(true) {(acc, hst) => (
+                          acc && (hst match {
+                            case x if 1 until 255 contains x.minValue().map(_.toInt).getOrElse(255) => true
```

Comment:
I misunderstood what clipping is intended to do on the upper end. I pushed a revision that should exhibit the correct behavior.